src/components/IconButton/index.tsx

```diff
@@ -0,0 +1,39 @@
+import styled from '@emotion/styled';
+
+import ArrowBackIcon from '@/assets/icons/arrow-back.svg?react';
+import FavoriteDefaultIcon from '@/assets/icons/favorite-default.svg?react';
+import SearchIcon from '@/assets/icons/search.svg?react';
+import StoreDefaultIcon from '@/assets/icons/store-default.svg?react';
+
+interface IconButtonProps {
+  icon: string;
```

Comment:
코드를 보면 현재 특정한 Icon만 사용이 가능한데요. 그런 경우에는 prop으로 넘길 때부터 적절한 아이콘을 넘길 수 있도록 타입을 좁혀주시면 좋을거 같아요
```suggestion
  icon: "arrow-back" | "favorite-default" | "search" | "store-default";
```

Reply:
조금 더 나아가면 render 될 icon 자체를 받는 식으로 구현하면 더 유연하게 이 컴포넌트를 사용할 수 있을거 같네요

Reply:
그렇게 타입을 일일이 지정해둘까 했는데, 
지금은 일부 아이콘만 넣어두었고, 아이콘이 더 많이 추가될 것 같아서 스트링으로 두었습니다. (어차피 default로 에러 처리해서요)
아이콘 자체를 프롭으로 받게 하면 사용하는 컴포넌트나 페이지마다 아이콘을 import 해야 하니 코드도 길어지고 번거롭다 생각했습니다. 

다시 생각해보니 제안 주신 대로 타입 지정하면 자동완성에 떠서 코드 입력 시 좀 더 간편할 것 같아서 요렇게 하겠습니다!